src/mds/Anchor.h

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_ANCHOR_H
+#define CEPH_ANCHOR_H
+
+#include <string>
+using std::string;
+
+#include "include/types.h"
+#include "mdstypes.h"
+#include "include/buffer.h"
+
+class Anchor {
+public:
+  inodeno_t ino;      // anchored ino
+  inodeno_t dirino;
+  std::string d_name;
+  __u8 d_type;
+  union {
+    mutable int nref;		// reference count
+    mutable mds_rank_t auth;	// reference count
```

Comment:
I'm baffled by this union. Is the Anchor now supposed to also store the authoritative rank? Why is that unioned with the reference count?